File: scratch/read_transcript.php
<?php

function print_steps_from($conv_id, $from_step = 270) {
    $path = "C:\\Users\\ACER\\.gemini\\antigravity\\brain\\{$conv_id}\\.system_generated\\logs\\transcript.jsonl";
    echo "\n=========================================\n";
    echo "=== STEPS FROM $from_step OF {$conv_id} ===\n";
    echo "=========================================\n";
    if (!file_exists($path)) {
        echo "File does not exist: {$path}\n";
        return;
    }
    
    $lines = file($path);
    if ($lines === false) {
        echo "Failed to read {$path}\n";
        return;
    }
    
    foreach ($lines as $line) {
        $data = json_decode($line, true);
        if (!$data) continue;
        
        $step = $data['step_index'] ?? 0;
        if ($step < $from_step) continue;
        
        $t = $data['type'] ?? '';
        $source = $data['source'] ?? '';
        $content = $data['content'] ?? '';
        
        if ($t === 'USER_INPUT') {
            echo "\n[Step {$step}] USER REQUEST:\n" . trim($content) . "\n";
        } elseif ($t === 'PLANNER_RESPONSE' && !empty($content)) {
            $trimmed = strlen($content) > 1000 ? substr($content, 0, 1000) . "\n... [TRUNCATED] ..." : $content;
            echo "\n[Step {$step}] MODEL RESPONSE:\n" . trim($trimmed) . "\n";
        }
    }
}

print_steps_from("626cf84c-af02-46c1-8ad9-982eca1fe92e", 270);

File: scratch/read_transcript.py
import json
import sys

sys.stdout.reconfigure(encoding='utf-8')

def print_conv(conv_id):
    path = f"C:\\Users\\ACER\\.gemini\\antigravity\\brain\\{conv_id}\\.system_generated\\logs\\transcript.jsonl"
    print(f"=== CONVERSATION {conv_id} ===")
    try:
        with open(path, 'r', encoding='utf-8') as f:
            for line in f:
                data = json.loads(line)
                step = data.get('step_index')
                t = data.get('type')
                content = data.get('content', '')
                if t == 'USER_INPUT':
                    print(f"\n[Step {step}] USER INPUT:\n{content}\n")
                elif t == 'PLANNER_RESPONSE' and len(content) > 0:
                    # Only print some summaries or key points to avoid huge text
                    lines = content.split('\n')
                    summary_lines = [l for l in lines if any(x in l.lower() for x in ['bug', 'station', 'absensi', 'nodemcu', 'esp32', 'rfid', 'response', 'parse', 'conclusion', 'perbaikan', 'fix', 'github', 'push'])]
                    if summary_lines:
                        print(f"[Step {step}] MODEL PLANNER (filtered):")
                        for sl in summary_lines[:15]:
                            print("  ", sl)
    except Exception as e:
        print(f"Error reading {conv_id}: {e}")

print_conv("91698023-f768-4a10-8fb2-44a93324d3ee")
print_conv("626cf84c-af02-46c1-8ad9-982eca1fe92e")

